vhost-user-backend/src/backend.rs

```diff
@@ -205,13 +208,17 @@ pub trait VhostUserBackendMut: Send + Sync {
     /// function.
     fn set_backend_req_fd(&mut self, _backend: Backend) {}
 
-    #[cfg(feature = "gpu-socket")]
     /// Set handler for communicating with the frontend by the gpu specific backend communication
     /// channel.
     ///
-    /// This method only exits when the crate feature gpu-socket is enabled, because this is only
-    /// useful for a gpu device.
-    fn set_gpu_socket(&mut self, gpu_backend: GpuBackend);
+    /// This function returns a `Result`, returning an error if the backend does not implement this
+    /// function.
+    fn set_gpu_socket(&mut self, _gpu_backend: GpuBackend) -> Result<()> {
+        Err(std::io::Error::new(
+            std::io::ErrorKind::Unsupported,
+            "back end does not support gpu socket",
```

Comment:
Small nit (`back end` -> `backend` and extra info)

```suggestion
            "Backend does not support set_gpu_socket() / VHOST_USER_GPU_SET_SOCKET",
```

Reply:
I don't know why, but other messages are using `back end` so I suggested continuing in that direction, but I also prefer `backend`.